validate/data.go

```diff
@@ -24,36 +33,59 @@ func EntryReader(r dsio.EntryReader) ([]jsonschema.ValError, error) {
 		return nil, fmt.Errorf("error allocating data buffer: %s", err.Error())
 	}
 
+	validated := false
+
 	err = dsio.EachEntry(r, func(i int, ent dsio.Entry, err error) error {
 		if err != nil {
 			return fmt.Errorf("error reading row %d: %s", i, err.Error())
 		}
+
 		err = buf.WriteEntry(ent)
 		if err != nil {
 			return fmt.Errorf("error writing row %d: %s", i, err.Error())
 		}
+		validated = false
+
+		if (i+1)%batchSize == 0 {
```

Comment:
Moving this check above the call to `WriteEntry` would simplify it by removing the need to add 1.

Reply:
It's purpose was to avoid doing a flush on the 0th iteration. I've just moved the guard for "is validated" into the flush function so it should be fine now.